### PR TITLE
Fix metrics index "undefined method `[]' for nil:NilClass"

### PR DESCRIPTION
**Pull request by oliverli**
## Description

Add check if `course.assessments.group("category_name").count().max()` is `nil` before trying to access array value, because that can be `nil` if an instructor just created a course (hence no categories) and tries to view the student metrics page

## Motivation and Context

Fixed that bug which we were made aware of by Chakiel

## How Has This Been Tested?

To replicate bug:
* create new course
* try to view student metrics page
Confirmed bug has been fixed after this change, with desired effects on the dropdown box using `@max_consecutive_assessments`, i.e. it is basically disabled

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation, which is located at [Autolab Docs](https://github.com/autolab/docs)
- [ ] I have updated the documentation accordingly, included in this PR

## Other issues / help required

If unsure, feel free to submit first and we'll help you along.

### CODE REVIEW

File: app/controllers/metrics_controller.rb [hunk 1]
```diff
@@ -4,7 +4,13 @@ def index
 		course = Course.find_by(name: params[:course_name])
 		@course_grace_days = course.grace_days
 		@num_assessments = course.assessments.count()
-		@max_consecutive_assessments = course.assessments.group("category_name").count().max()[1] - 1
+		
+		course_max = course.assessments.group("category_name").count().max()
+		if (course_max == nil)
+			@max_consecutive_assessments = 0
+		else
+			@max_consecutive_assessments = course.assessments.group("category_name").count().max()[1] - 1
```
Comment (ashleyzhang) [inline app/controllers/metrics_controller.rb:12]: nit: could we change `course.assessments.group("category_name").count().max()` to just `course_max`?

**Review by ashleyzhang — COMMENTED** [review]
Small nit but looks good otherwise!

File: app/controllers/metrics_controller.rb (hunk 1 above)
Comment (oliverli) [reply app/controllers/metrics_controller.rb]: good catch! done

**Review by ashleyzhang — APPROVED** [review]
LGTM! Thanks for the fix

**Pull request merged**